docs/modules/ROOT/pages/how-to-guides/Secure-your-supply-chain/proc_java_dependencies.adoc

```diff
@@ -1,30 +1,110 @@
+
+- When the system has completed all its builds, you can rebuild your application.
+
+- If all the builds are successful, your application's SBOM displays that none of the dependencies came from third-party repositories. However, you may need to troubleshoot when some dependencies fail to build.
+
+== Setup
+
+=== Logging into the Server
+
+All interactions with the JVM Build Service require you to be logged into the OpenShift namespace used for your builds.
+
+.Procedure
+
+Copy the login command from: https://registration-service-toolchain-host-operator.apps.stone-prd-host1.wdlc.p1.openshiftapps.com/
+
+=== Setting up the CLI
+
+Many of the commands below can be setup using the JVM Build Service CLI.
+
+The CLI is currently distributed as a docker image. To use the docker image, create the following alias:
+
+```
+alias jbs='docker run --mount type=bind,source=$HOME/.kube/config,target=/kube --mount type=bind,source=$HOME/.github,target=/root/.github --env KUBECONFIG=/kube  -it --rm   quay.io/redhat-appstudio/hacbs-jvm-cli:latest'
+```
+
+To update the latest version, create an update alias. You can run this alias to pull the latest version of the docker image:
+
+```
+alias update-jbs='docker  pull quay.io/redhat-appstudio/hacbs-jvm-cli:latest'
+```
+
+The CLI supports tab completion and all the commands have a --help option to show usage.
+
+Note that this CLI also requires you to be logged in as described above.
+
+To use commands that modify build recipes (i.e. commands that are used to fix failing builds) you will need to create a `$HOME/.github` file as specified at  https://github-api.kohsuke.org/[].
+
+== Configuring The JVM Rebuild Service
+
+There are three ways to setup the JVM Build Service, they are listed below.
+
+=== Setting up using the CLI
+
+If you do not require explicit quay.io configuration, and you have setup the CLI this is the simplest way to setup the JVM Build Service. To do this run the `setup rebuilds` command in the CLI:
+
+[source]
+----
+jbs> setup rebuilds
+Working...
+found .dockerconfigjson secret with appropriate token keys in namespace sdouglas1-tenant, rebuilds are possible
+Rebuilds setup successfully
+----
+
+This will automatically perform the steps detailed below and setup rebuilds for you. The JVM Build Service will automatically create a quary repository to store your rebuilt artifacts.
+
+=== Manual Setup Without Quay.io Configuration*
+
+If you do not want to use the CLI this method creates the Kubernetes objects to setup the JVM Build Service directory using `kubectl`.
+
+. Create a file, for example, `config.yaml`.
+
+. In the `config.yaml` file, create a `JBSConfig` resource with the following data:
+
++
+[source,yaml]
+----
+apiVersion: jvmbuildservice.io/v1alpha1
+kind: JBSConfig
+metadata:
+  name: jvm-build-config
+spec:
+  enableRebuilds: "true"
+----
+
+. Run `kubectl apply -f config.yaml`
+
+
+=== Manual Procedure With Explicit Quay.io Config
```

Comment:
--> follow same heading convention as the other 2 with gerund verbs and explicitly saying what the user is doing here. "Setting up the ______ using the quay.io configuration"?